crates/net/network-api/src/downloaders.rs

```diff
@@ -0,0 +1,16 @@
+//! API related to syncing blocks.
+
+use futures::Future;
+use reth_network_p2p::BlockClient;
+use tokio::sync::oneshot;
+
+/// Provides client for downloading blocks.
+#[auto_impl::auto_impl(&, Arc)]
+pub trait BlockDownloaderProvider {
+    /// Returns a new [`BlockClient`], used for fetching blocks from peers.
+    ///
+    /// The client is the entrypoint for sending block requests to the network.
+    fn fetch_client(
+        &self,
+    ) -> impl Future<Output = Result<impl BlockClient + 'static, oneshot::error::RecvError>> + Send;
```

Comment:
pretty sure we can make the Blockclient trait 'static

Reply:
better to avoid it because otherwise we will have some problem in future the we have in eth api helper traits for returning opaque `ConfigureEvm` type: that we have to explicitly code `-> &impl ConfigureEvm` because `-> impl ConfigureEvm` isn't static if the reference doesn't outlive the ref to self in signature

```rust
fn my_method(&self) -> &impl ConfigureEvm;
```